### PR TITLE
TYP: more annotations for io.pytables

### PR DESCRIPTION
**Pull request by jbrockmendel**
should be orthogonal to #29692.

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -1123,7 +1142,7 @@ def append_to_multiple(
 
         # figure out how to split the value
         remain_key = None
-        remain_values = []
+        remain_values: List = []
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Any chance of adding a subtype here?

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -1077,7 +1089,14 @@ def append(
         self._write_to_group(key, value, append=append, dropna=dropna, **kwargs)
 
     def append_to_multiple(
-        self, d, value, selector, data_columns=None, axes=None, dropna=False, **kwargs
+        self,
+        d: Dict,
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Same comment

**Review by WillAyd — COMMENTED** [review]
Looks good. A few non-blockers that could be improved otherwise @simonjayhawkins

File: pandas/io/pytables.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: same here and below: i'm pretty sure we'll be able to be more specific in these after #29692

**Comment by jreback** [comment]
thanks

**Pull request merged**